### ISSUE TITLE
Bring back the integrated debugger

### ISSUE THREAD

**Issue by codeanticode** [processing/processing-android#686]
The integrated debugger is currently disabled due to the changes in the API of the PDE editor in Processing 4:

https://github.com/processing/processing-android/blob/master/mode/src/processing/mode/android/AndroidEditor.java#L88

**Comment by rupeshkumar22** [comment]
Some progress has been made here - https://github.com/processing/processing-android/compare/master...rupeshkumar22:processing-android:debuggerImprovements

the above changes were not sufficient to make it run, adding android:debuggable="true" in the manifest before running the sketch on the emulator was essential to resolve an issue mentioned here - https://stackoverflow.com/questions/18893656/application-is-waiting-for-the-debugger-to-attach

For now, Debugger seems to be working fine-
![DebuggerUpAndRunning](https://user-images.githubusercontent.com/46577873/184463800-d9ed57e1-20e4-4db9-8b13-477a0500cf1c.PNG)

I need some information to test the working of the debugger. Is there any documentation available to test the debugger's functionality? @codeanticode @ranaaditya 
Thanks!

**Comment by rupeshkumar22** [comment]
@codeanticode @ranaaditya  I am looking for versions of Android Mode and processing 3 in which the debugger was stable and working well last time when @manav-mj contributed. Do you remember the versions @codeanticode ? I am able to launch and run the debugger but some null pointer exceptions were faced when trying to use **Continue, Step** buttons. Comparing the changes from the stable debugger branch would be helpful. Currently, I'm going through commit #491 to understand the debugger workflow.

**Comment by codeanticode** [comment]
The debugger should work with version 4.3.0 of the Android mode and version 3.5.4 of Processing.

**Comment by rupeshkumar22** [comment]
Thanks for informing the versions @codeanticode 
A recent catch-up with @ranaaditya has been helpful to understand the debugger and some other functionalities.
I've reached a stage where AndroidDebugger's step, continue and other button functionalities are working fine. For variable-value window data reflection, we have started using VariableInspector.java in processing4. I have integrated the use of the VariableInspector in the debugger but data is not being reflected in the variable-value window for some reasons. I'm looking into it. Probably, I'll find a solution for the same.
Thanks
